tests/test_dockerfile.py

```
@@ -0,0 +1,143 @@
+        "WORKDIR should be /app"
+
+
+def test_single_workdir_line(dockerfile_text: str) -> None:
+    count = len(re.findall(r"^\s*WORKDIR\b", dockerfile_text, re.M))
+    assert count == 1, "Expect exactly one WORKDIR instruction"
+
+
+def test_copy_requirements_then_pip_install_then_copy_source(dockerfile_text: str) -> None:
+    insns = _instructions(dockerfile_text)
+
+    def find_index(predicate: Callable[[str, str], bool]) -> Optional[int]:
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)